contracts/Library/RequestScheduleLib.sol

```diff
@@ -0,0 +1,232 @@
+     */
+    function firstClaimBlock(ExecutionWindow storage self) 
+        internal view returns (uint)
+    {
+        return freezeStart(self).sub(self.claimWindowSize);
+    }
+
+    /*
+     *  Helper: Returns boolean if we are before the execution window.
+     */
+    function isBeforeWindow(ExecutionWindow storage self)
+        internal view returns (bool)
+    {
+        return getNow(self) < self.windowStart;
+    }
+
+    /*
+     *  Helper: Returns boolean if we are after the execution window.
+     */
+    function isAfterWindow(ExecutionWindow storage self) 
+        internal view returns (bool)
+    {
+        return getNow(self) > windowEnd(self);
+    }
+
+    /*
+     *  Helper: Returns boolean if we are inside the execution window.
+     */
+    function inWindow(ExecutionWindow storage self)
+        internal view returns (bool)
+    {
+        return self.windowStart <= getNow(self) && getNow(self) < windowEnd(self);
+    }
+
+    /*
+     *  Helper: Returns boolean if we are inside the reserved portion of the
+     *  execution window.
+     */
+    function inReservedWindow(ExecutionWindow storage self)
+        internal view returns (bool)
+    {
+        return self.windowStart <= getNow(self) && getNow(self) < reservedWindowEnd(self);
+    }
+
+    /*
+     * @dev Helper: Returns boolean if we are inside the claim window.
+     */
+    function inClaimWindow(ExecutionWindow storage self) 
+        internal view returns (bool)
+    {
+        /// Checks that the firstClaimBlock is in the past or now.
+        /// Checks that now is before the start of the freezePeriod.
+        if (firstClaimBlock(self) <= getNow(self)
+            && getNow(self) < freezeStart(self)) {
+                return true;
+        }
+        return false;
+    }
+
+    /*
+     *  Helper: Returns boolean if we are before the freeze period.
+     */
+    function isBeforeFreeze(ExecutionWindow storage self) 
+        internal view returns (bool)
+    {
+        return getNow(self) < freezeStart(self);
+    }
+
+    /*
+     *  Helper: Returns boolean if we are before the freeze period.
+     */
+    function isBeforeClaimWindow(ExecutionWindow storage self)
+        internal view returns (bool)
+    {
+        return getNow(self) < firstClaimBlock(self);
+    }
+
+    ///---------------
+    /// VALIDATION
+    ///---------------
+
+    /**
+     * @dev Validation: Ensure that the reservedWindowSize is less than or equal to the windowSize.
+     * @param _reservedWindowSize The size of the reserved window.
+     * @param _windowSize The size of the execution window.
+     * @return True if the reservedWindowSize is within the windowSize.
+     */
+    function validateReservedWindowSize(uint _reservedWindowSize,
+                                        uint _windowSize)
+        public pure returns (bool)
+    {
+        return _reservedWindowSize <= _windowSize.add(1); //? - why add 1?
```

Comment:
I can't figure out why `add(1)`.  I suspect it was some sort of sanity-check or validation but I can't figure out what.  If you remove it do the tests fail?

Reply:
I couldn't figure out its purpose either and the tests run fine with it taken out.